### PR TITLE
allow certain users to be ignored from issue lookups

### PR DESCRIPTION
**Pull request by jlambert121**

### CODE REVIEW

**Comment by amaltson** [comment]
Thanks for the PR @jlambert121, from a quick glance it looks good. I'll take a more thorough look tonight. Thanks!

File: lib/lita/handlers/jira_issues.rb [hunk 1]
```diff
@@ -8,15 +8,18 @@ class JiraIssues < Handler
...
 
       route /[a-zA-Z]+-\d+/, :jira_message, help: {
         "KEY-123" => "Replies with information about the given JIRA key"
       }
 
       def jira_message(response)
-        @jira ||= JiraGateway.new(http, config)
-        Set.new(response.matches).each do | key |
-          handle_key(response, key)
+        if !config.ignore.include?(response.user.name)
```
Comment (amaltson) [inline lib/lita/handlers/jira_issues.rb]: I like the [Confident Ruby](http://www.confidentruby.com/) style guards, so something like:

```ruby
def jira_message(response)
  return if config.ignore.include?(response.user.name)
  ...
end
```

**Comment by amaltson** [comment]
Very minor nit pick, if you can send over the change that'd be great. If not, I can definitely merge this in and make the change myself.

If I don't hear back in a day I'll merge in, update the README and release a new version. Thanks again for your PR!

**Comment by jlambert121** [comment]
Updated - thanks for pointing out where your preference came from, I hadn't seen that style guide before but am going to check it out.

File: lib/lita/handlers/jira_issues.rb [hunk 2]
```diff
@@ -8,15 +8,18 @@ class JiraIssues < Handler
...
 
       route /[a-zA-Z]+-\d+/, :jira_message, help: {
         "KEY-123" => "Replies with information about the given JIRA key"
       }
 
       def jira_message(response)
-        @jira ||= JiraGateway.new(http, config)
-        Set.new(response.matches).each do | key |
-          handle_key(response, key)
+        return if config.ignore.include?(response.user.name)
```
Comment (amaltson) [inline lib/lita/handlers/jira_issues.rb]: Sorry, that's an inline if statement, so the lines below should be at the same indentation (and there's an extra end). It should look something like:

```ruby
def jira_message(response)
  return if config.ignore.include?(response.user.name)
  @jira ||= JiraGateway.new(http, config)
  Set.new(response.matches).each do | key |
    handle_key(response, key)
  end
end
```

**Comment by amaltson** [comment]
No worries @jlambert121, that's one of my favourite programming books, at about the same level as [Eloquent Ruby](http://www.amazon.com/Eloquent-Ruby-Addison-Wesley-Professional-Series/dp/0321584104).

**Comment by jlambert121** [comment]
The indentation I should have caught, sorry about that.  Updated.  Thanks for the other ref as well.

**Comment by amaltson** [comment]
:ship: it. I'll update the README and push up a new version tonight. Thanks for the PR (my first ever :))

**Pull request merged**